extensions/airship/airship-event-extension.schema.json

```diff
@@ -0,0 +1,295 @@
+    },
+    "https://ns.airship.com/device": {
+      "title": "Airship Device",
+      "description": "Contains Airship specific device attributes.",
+      "type": "object",
+      "properties": {
+        "https://ns.airship.com/attributes": {
+          "title": "Airship Device Attributes",
+          "description": "Attributes specific to devices in Airship.",
+          "type": "object",
+          "properties": {
+            "https://ns.airship.com/localeVariant": {
+              "title": "Airship Locale Variant",
+              "description": "The language locale variant if one is reported for the device..",
+              "type": "string"
+            },
+            "https://ns.airship.com/uaSdkVersion": {
+              "title": "Airship SDK Version",
+              "description": "The version of the Airship SDK used in the app.",
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "https://ns.airship.com/body": {
+      "title": "Airship Event Body",
+      "description": "Contains Airship specific event body fields.",
+      "type": "object",
+      "properties": {
+        "https://ns.airship.com/pushId": {
+          "title": "Push ID",
+          "description": "A unique identifier for a push operation.",
+          "type": "string"
+        },
+        "https://ns.airship.com/groupId": {
+          "title": "Airship Group ID",
+          "description": "Identifies a push delivered over an interval of time, e.g. multiple push_ids as part of the fulfillment of an automation or a push-to-local-time specification.",
+          "type": "string"
+        },
+        "https://ns.airship.com/variantId": {
+          "title": "Airship Variant ID",
+          "description": "The ID of the variant that a push is associated with, if the push was a part of an A/B test (experiment).",
+          "type": "string"
+        },
+        "https://ns.airship.com/alerting": {
+          "title": "Alerting Push",
+          "description": "If true, the send event was alerting. Alerting send event has notification text, badge, or sound.",
+          "type": "boolean"
+        },
+        "https://ns.airship.com/triggeringPush": {
+          "title": "Triggering Push",
+          "description": "The specific push notification details associated with an event. An associated push helps you trace an event to the original notification or operation.An associated push object may specify a `time`, if the push was a singular operation sent at a defined time. Otherwise, the object will include a `group_id` if the push was sent at a relative time (`best_time` or `local_time`) an automation pipeline, or another operation resulting in multiple `push_id`s.",
+          "type": "object",
+          "properties": {
+            "https://ns.airship.com/campaign": {
```

Comment:
As you are using campaign object multiple times/places I suggest you to create Campaign data type and reference that data type where and when needed

Reply:
Maybe as a local definition that is referenced, but lets please not introduce yet another datatype to represent an Airship Campaign that is never to be used outside of this mixin. Keep everything contained in this one file.

Reply:
That's correct, it's the same Campaign object in multiple places but I wasn't sure if there's a better way to handle it.